FlowCrypt/Models/Realm Models/KeypairRealmObject.swift

```diff
@@ -45,25 +38,23 @@ extension KeypairRealmObject {
     convenience init(_ keyDetails: KeyDetails, passphrase: String?, source: KeySource, user: UserRealmObject) throws {
         self.init()
 
-        guard let privateKey = keyDetails.private else {
-            throw KeyInfoError.missingPrivateKey("storing pubkey as private")
+        guard let privateKey = keyDetails.private, keyDetails.isFullyEncrypted != nil else {
```

Comment:
Is `keyDetails.isFullyEncrypted != nil` check necessary here? As we check `keyDetails.isFullyEncrypted == true` at below line?
And maybe we need to take care of localizations? Error message is returned as static string here.

Reply:
Agree with localizations

nil is different than false. nil means it's not a private key. false means it's a private key that is not encrypted fully. Notice nil and false produces a different error, as implemented.

Reply:
👍

Reply:
In that case we may update code like this.

```
guard let privateKey = keyDetails.private, let isFullyEncrypted = keyDetails.isFullyEncrypted else {
    throw KeypairError.missingPrivateKey("storing pubkey as private")
}
guard isFullyEncrypted else {
    throw KeypairError.notEncrypted("Will not store Private Key that is not fully encrypted")
}
```

Reply:
either is fine with me